### PR TITLE
Allow passing `name` as kwarg to pm.Bound RVs

### LINKED ISSUE #3149: Cannot apply bound to Poisson distribution
If you have questions about a specific use case, or you are not sure whether this is a bug or not, please post it to our discourse channel: https://discourse.pymc.io  

When I apply Bound to the Poisson I get the following error:

**Please provide a minimal, self-contained, and reproducible example.**
```python
import pymc3 as pm

example = pm.Model()

with example:
    BoundPoisson = pm.Bound(pm.Poisson, upper = 6)
    y = BoundPoisson(name = "y", mu = 1)
```

**Please provide the full traceback.**
```python
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-18-c9645cb7d458> in <module>()
      3 with example:
      4     BoundPoisson = pm.Bound(pm.Poisson, upper = 6)
----> 5     y = BoundPoisson(name = "y", mu = 1)

~/miniconda3/lib/python3.6/site-packages/pymc3/distributions/bound.py in __call__(self, *args, **kwargs)
    209                              'with the cumulative probability function. See '
    210                              'pymc3/examples/censored_data.py for an example.')
--> 211         first, args = args[0], args[1:]
    212 
    213         if issubclass(self.distribution, Continuous):

IndexError: tuple index out of range
```

**Please provide any additional information below.**

This does not occur when I apply Bound to Normal, which makes me believe that this issue only applies to Poisson, or possibly other discrete distributions.

## Versions and main components

* PyMC3 Version: 3.5
* Theano Version: 1.0.2
* Python Version: 3.6.2
* Operating system: Ubuntu 16.04
* How did you install PyMC3: (conda/pip) pip

### PR DESCRIPTION
**Pull request by aseyboldt**
fix #3149.

### CODE REVIEW

File: pymc3/tests/test_distributions.py [hunk 1]
```diff
@@ -1183,6 +1183,10 @@ def test_bound():
     assert rand.dtype in [np.int16, np.int32, np.int64]
     assert rand >= 5 and rand <= 8
 
+    with pm.Model():
+        BoundPoisson = pm.Bound(pm.Poisson, upper=6)
+        BoundPoisson(name="y", mu=1)
```
Comment (junpenglao) [inline pymc3/tests/test_distributions.py]: Hmmmmmmmmm we never do this... it is always `BoundPoisson("y", mu=1)`

Comment (junpenglao) [reply pymc3/tests/test_distributions.py]: (You must have been playing around with TFP and Edwards2 lately LOL)

Comment (aseyboldt) [reply pymc3/tests/test_distributions.py]: Wasn't me :-)
I think an explicit `name` should work though. That is just the norm in python.

Comment (junpenglao) [reply pymc3/tests/test_distributions.py]: I see. We do have the same pattern in FreeRV etc

Comment (aseyboldt) [reply pymc3/tests/test_distributions.py]: This works:
```
with pm.Model() as model:
    pm.Normal(name='a')
    pm.Normal(name='b', observed=[0.])
```

**Pull request merged**